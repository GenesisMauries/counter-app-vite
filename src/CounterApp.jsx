import PropTypes from "prop-types";
// const handleAdd = (event) => console.log("Mas uno", event); // Fuera porque no tiene nada que ver con el componente y para que no se le llame cada que renderiza el componente
export const CounterApp = ({ value }) => {
  const handleAdd = (event, value) => console.log("Mas uno", event, value);
  return (
    <>
      <h1>Counter App</h1>
      <h2> {value} </h2>
      {/* Evento */}
      {/* <button onClick={handleAdd}>+1</button> */}
      {/* Evento + de un argumento */}
      <button onClick={(event) => handleAdd(event, value)}>+1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
