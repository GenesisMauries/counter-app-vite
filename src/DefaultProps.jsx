import PropTypes from 'prop-types';
export const DefaultProps = ({ title, subTitle }) => {
  return (
    <>
      <h4>Default Props Example</h4>
      <p data-testid='test-title'>{title}</p>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <hr />
    </>
  );
};

// Define el tipo de datos
DefaultProps.propTypes = {
  // title: PropTypes.number,
  title: PropTypes.string.isRequired, // Vuelve obligatorio este prop
  subTitle: PropTypes.string,
};

DefaultProps.defaultProps = {
  subTitle: 'No envias subtitulo y este string es un prop por defecto',
};
