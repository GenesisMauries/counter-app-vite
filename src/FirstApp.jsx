// si no esta alterando el estado va fuera del componente jsx
// para no requerirlo cada que se monta
const name = "GÃ©nesis";

// Fuera del componente para no usar espacio en memoria
const multiplicacion = (a, b) => {
  return a * b;
};

export const FirstApp = () => {
  return (
    <>
      <h4>Retornar un grupo de elementos HTML</h4>
      <p>Con un fragment</p>
      <p>para no romper el estilo con un elemento div/section</p>
      {/*  No es recomendable mostrar de esta forma el resultado de la fn */}
      <p> Multiplicacion {multiplicacion(39, 11)}</p>
      <i>by {name}</i>
      <hr />
    </>
  );
};
