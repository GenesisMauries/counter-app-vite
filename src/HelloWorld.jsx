export const HelloWorld = () => {
  return <h4>Hola Mundaa</h4>;
};
