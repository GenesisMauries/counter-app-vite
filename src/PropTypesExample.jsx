import PropTypes from "prop-types";
export const PropTypesExample = ({ title, address }) => {
  return (
    <>
      <h4>PropsTypes Example</h4>
      <p>{title}</p>
      <p>{address}</p>
      <hr />
    </>
  );
};

// Define el tipo de datos
PropTypesExample.propTypes = {
  // title: PropTypes.number,
  title: PropTypes.number.isRequired, // Vuelve obligatorio este prop
  address: PropTypes.string,
};
