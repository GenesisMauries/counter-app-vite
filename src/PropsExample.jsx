export const PropsExample = ({ title, subTitle, number }) => {
  // desestructura objeto props
  // props sirve para enviar del padre al hijo
  // console.log(props);
  return (
    <>
      <h2>Props Example</h2>
      <h4>{title}</h4>
      <h4>{subTitle}</h4>
      <h4>Soy un n√∫mero {number}</h4>
      <hr />
    </>
  );
};
