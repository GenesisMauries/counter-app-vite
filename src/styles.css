html,
body {
  background-color: #21232a;
  color: #fff;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1.3em;
  padding: 70px;
}
button {
  padding: 5px;
  font-size: 1.3em;
  border-radius: 5px;
  margin-left: 8px;
}
