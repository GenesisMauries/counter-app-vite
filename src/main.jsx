import React from 'react';
import { createRoot } from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { DefaultProps } from './DefaultProps';
import { FirstApp } from './FirstApp';
import { HelloWorld } from './HelloWorld';
import { PropsExample } from './PropsExample';
import { PropTypesExample } from './PropTypesExample';

// Definir estilos
import './styles.css';

createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <CounterApp value={10} />
    <HelloWorld /> */}
    {/* <FirstApp /> */}

    {/* Envia props del padre al hijo */}
    <PropsExample
      title='Vengo de props'
      subTitle='Me envian desde main.jsx'
      number={1000}
    />
    {/* 
    <PropTypesExample title={123} address='Coyoacan' />
    <DefaultProps title={123} /> */}
  </React.StrictMode>
);
