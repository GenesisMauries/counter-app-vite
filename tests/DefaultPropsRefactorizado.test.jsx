import { render, screen } from '@testing-library/react';
import { DefaultProps } from '../src/DefaultProps';

describe('Pruebas <DefaultProps7>', () => {
  // Hace una captura del componente y la almacena en la carpeta __snapshots__
  const title = '123';
  const subTitle = 'Subtitulo';
  test('Debe hacer match con snapshot', () => {
    const { container } = render(<DefaultProps title={title} />);
    expect(container).toMatchSnapshot();
  });
  test('Debe mostrar el mensaje 123', () => {
    // screen.debug(); // screen antes de montar el componente
    render(<DefaultProps title={title} />);
    // screen.debug(); // screen despues de montar/render el componente
    expect(screen.getByText(title)).toBeTruthy();
  });
  test('Debe mostrar el subtitulo enviado por props', () => {
    render(<DefaultProps title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle)).toBeTruthy();
  });
});
