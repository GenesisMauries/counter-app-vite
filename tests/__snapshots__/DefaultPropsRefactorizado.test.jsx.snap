// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <DefaultProps/> Debe hacer match con snapshot 1`] = `
<div>
  <h4>
    Default Props Example
  </h4>
  <p
    data-testid="test-title"
  >
    123
  </p>
  <p>
    No envias subtitulo y este string es un prop por defecto
  </p>
  <p>
    No envias subtitulo y este string es un prop por defecto
  </p>
  <p>
    No envias subtitulo y este string es un prop por defecto
  </p>
  <hr />
</div>
`;

exports[`Pruebas <DefaultProps7> Debe hacer match con snapshot 1`] = `
<div>
  <h4>
    Default Props Example
  </h4>
  <p
    data-testid="test-title"
  >
    123
  </p>
  <p>
    No envias subtitulo y este string es un prop por defecto
  </p>
  <p>
    No envias subtitulo y este string es un prop por defecto
  </p>
  <p>
    No envias subtitulo y este string es un prop por defecto
  </p>
  <hr />
</div>
`;
